data/src/main/java/org/apache/iceberg/data/TableMigrationUtil.java

```diff
@@ -74,111 +74,151 @@ private TableMigrationUtil() {
   public static List<DataFile> listPartition(Map<String, String> partition, String uri, String format,
                                              PartitionSpec spec, Configuration conf, MetricsConfig metricsConfig,
                                              NameMapping mapping) {
+    int parallelism = conf.getInt(PARQUET_READ_PARALLELISM, 1);
     if (format.contains("avro")) {
-      return listAvroPartition(partition, uri, spec, conf);
+      return listAvroPartition(partition, uri, spec, conf, parallelism);
     } else if (format.contains("parquet")) {
-      return listParquetPartition(partition, uri, spec, conf, metricsConfig, mapping);
+      return listParquetPartition(partition, uri, spec, conf, metricsConfig, mapping, parallelism);
     } else if (format.contains("orc")) {
-      return listOrcPartition(partition, uri, spec, conf, metricsConfig, mapping);
+      return listOrcPartition(partition, uri, spec, conf, metricsConfig, mapping, parallelism);
     } else {
       throw new UnsupportedOperationException("Unknown partition format: " + format);
     }
   }
 
   private static List<DataFile> listAvroPartition(Map<String, String> partitionPath, String partitionUri,
-                                                  PartitionSpec spec, Configuration conf) {
+                                                  PartitionSpec spec, Configuration conf, int parallelism) {
     try {
       Path partition = new Path(partitionUri);
       FileSystem fs = partition.getFileSystem(conf);
-      return Arrays.stream(fs.listStatus(partition, HIDDEN_PATH_FILTER))
-          .filter(FileStatus::isFile)
-          .map(stat -> {
-            InputFile file = HadoopInputFile.fromLocation(stat.getPath().toString(), conf);
-            long rowCount = Avro.rowCount(file);
-            Metrics metrics = new Metrics(rowCount, null, null, null, null);
-            String partitionKey = spec.fields().stream()
+      List<FileStatus> fileStatus = Arrays.stream(fs.listStatus(partition, HIDDEN_PATH_FILTER))
```

Comment:
This code (up until "task.run") is now identical amongst the three read paths. I think we should probably refactor the three functions now, since they really are only different in the "task.run" portion of the function. Everything else should be the same

Reply:
For more clarity I think it may be cleaner as 
```java
listPartition() {
  Read Files statuses
  Build Task Builder
  if {parquet} {
     task.run -> parquet code
  } ...
  
  ```

Reply:
OK

Reply:
Yeah I think this is a good path